### PR TITLE
Create PSC tokens during OVA init

### PR DESCRIPTION
**Pull request by anchal-agrawal**
Fixes vmware/vic#5551

### CODE REVIEW

File: installer/packer/scripts/systemd/psc/get_token.service [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=OVA PSC Token Creation
+Documentation=http://github.com/vmware/vic-product/installer
+After=docker.service systemd-networkd.service systemd-resolved.service iptables.service ova-firewall.service admiral_startup.path
```
Comment (anchal-agrawal) [inline installer/packer/scripts/systemd/psc/get_token.service]: @andrewtchin looking for feedback here, thanks!

File: installer/packer/scripts/psc/put_token_guestinfo.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script grabs the engine PSC token and puts it in guestinfo.
+set -euf -o pipefail
+
+engine_psc_token=$(grep "access_token" /etc/vmware/psc/engine/tokens.properties | awk -F'=' '{print $2}')
```
Comment (hmahmood) [inline installer/packer/scripts/psc/put_token_guestinfo.sh]: You could just merge this into `get_token.sh`, and get rid of the extra systemd unit.

Comment (anchal-agrawal) [inline installer/packer/scripts/psc/put_token_guestinfo.sh]: Is there a better way to do this? Specially if `=` may be present in the token itself.

Comment (anchal-agrawal) [inline installer/packer/scripts/psc/put_token_guestinfo.sh]: Having a separate script for putting the token in guestinfo would be useful while refreshing tokens. Thoughts?

Comment (anchal-agrawal) [inline installer/packer/scripts/psc/put_token_guestinfo.sh]: Fixed!

File: installer/packer/scripts/systemd/psc/get_token.service (hunk 1 above)
Comment (andrewtchin) [inline installer/packer/scripts/systemd/psc/get_token.service]: Does the admiral service need to be up all the way before you run get tokens? If so, it should be `admiral.service`
I think you can probably remove all the other units from `After` besides the correct admiral one

File: installer/packer/scripts/systemd/psc/get_token.service [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+Before=harbor_startup.service
+
+[Service]
+Type=oneshot
+ExecStartPre=/usr/bin/chmod +x /etc/vmware/psc/get_token.sh
```
Comment (andrewtchin) [inline installer/packer/scripts/systemd/psc/get_token.service]: Is `get_token.sh` checked in with `x` perms? Did it not propagate when copied by Packer?

File: installer/packer/scripts/system_settings.sh [hunk 4]
```diff
@@ -31,6 +31,7 @@ systemctl enable ovf-network.service
 systemctl enable ova-firewall.service
 systemctl enable harbor_startup.path
 systemctl enable admiral_startup.path
+systemctl enable get_token.service
```
Comment (andrewtchin) [inline installer/packer/scripts/system_settings.sh]: If we need to wait for the user to enter credentials before running this unit, you can remove this line.

**Comment by andrewtchin** [comment]
Where does the getting the token process use the admin credentials from the landing page?

**Comment by anchal-agrawal** [comment]
> Where does the getting the token process use the admin credentials from the landing page?

@andrewtchin Only the `register` command needs user creds. The `get-token` command uses the config file generated by the `register` command.

File: installer/packer/scripts/system_settings.sh (hunk 4 above)
Comment (anchal-agrawal) [inline installer/packer/scripts/system_settings.sh]: Good call - `get_token.service` doesn't need to start at boot time. It's started explicitly by `admiral_startup.service`.

File: installer/packer/scripts/systemd/psc/get_token.service (hunk 3 above)
Comment (anchal-agrawal) [inline installer/packer/scripts/systemd/psc/get_token.service]: It doesn't have the `x` bit set locally - I'll update it.

File: installer/packer/scripts/systemd/psc/get_token.service (hunk 1 above)
Comment (anchal-agrawal) [inline installer/packer/scripts/systemd/psc/get_token.service]: Re: ordering of getting tokens and admiral startup - the details aren't final yet, but for now the plan is to start admiral after the PSC config file  is present (`admiral_startup.path`).

I'll remove the other `After` deps.

**Pull request merged**